### PR TITLE
Add searchbox component 

### PR DESCRIPTION
**Pull request by prodhan29**
- create searchbox component that will be used in household list.
- add style for it
- add test code for it

### CODE REVIEW

File: src/components/page/SearchBox/SearchBox.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export interface SearchBoxState {
+  searchText: string;
+}
+
+/** default state for searchBox */
+const defaultSearchBoxState = {
+  searchText: '',
+};
+
+class SearchBox extends React.Component<SearchBoxProps, SearchBoxState> {
```
Comment (p-netm) [inline src/components/page/SearchBox/SearchBox.tsx]: could we add a doc-string here

File: src/components/page/SearchBox/SearchBox.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            className="search-bar-input"
+            onChange={this.handleChange}
+            onKeyPress={this.search}
+          />
+        </InputGroup>
+      </div>
+    );
+  }
+
+  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (p-netm) [inline src/components/page/SearchBox/SearchBox.tsx]: a doc-string for this

File: src/components/page/SearchBox/SearchBox.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return this.setState({ searchText: event.target.value });
+  };
+
+  private search = (event: React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (p-netm) [inline src/components/page/SearchBox/SearchBox.tsx]: And a doc-string for this.

File: src/components/page/SearchBox/SearchBox.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            onChange={this.handleChange}
+            onKeyPress={this.search}
+          />
+        </InputGroup>
+      </div>
+    );
+  }
+
+  private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    return this.setState({ searchText: event.target.value });
```
Comment (p-netm) [inline src/components/page/SearchBox/SearchBox.tsx]: does not seem like we should be returning anything here.

File: src/components/page/SearchBox/SearchBox.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  }
+  public render() {
+    return (
+      <div id="custom-search-input">
+        <InputGroup className="shadow-sm">
+          <span className="input-group-btn search-bar-icon">
+            <FontAwesomeIcon icon={['fas', 'search']} />
+          </span>
+          <Input
+            placeholder={this.props.placeholder || 'Search'}
```
Comment (p-netm) [inline src/components/page/SearchBox/SearchBox.tsx]: nitpick: you can define the component's `defaultProps` and have this just as `this.props.placeholder`.

**Comment by p-netm** [comment]
Also can we give the component a maximum width.

File: src/components/page/SearchBox/SearchBox.tsx (hunk 1 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: doc-string added

File: src/components/page/SearchBox/SearchBox.tsx (hunk 5 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: Added default props.

File: src/components/page/SearchBox/SearchBox.tsx (hunk 1 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: doc-string added

File: src/components/page/SearchBox/SearchBox.tsx (hunk 2 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: doc-string added

File: src/components/page/SearchBox/SearchBox.tsx (hunk 3 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: doc-string added

File: src/components/page/SearchBox/SearchBox.tsx (hunk 4 above)
Comment (prodhan29) [reply src/components/page/SearchBox/SearchBox.tsx]: Remove the return word

**Pull request merged**